include/curl-session.h

```diff
@@ -78,7 +78,8 @@ class CurlSession : public CurlCallbacks
     }
 
 
-    void addFormFile(const std::string& key, const std::string& filePath, const std::string& mimeType);
+    void addFormFile(const char* key, const char* filePath, const char* mimeType);
```

Comment:
One should you std::string unless there is a good reason not to do it. My point is that allocating/copying/deallocating memory on an (embedded) camera is the thing to avoid and if I know I have many calls that just pass const char* I'm accounting for it.
Of course, to get the most from this approach, it should be made everywhere. I just illustrated the point in one place. I would consider this an optimization and would agree it may be too early for such kind optimizations.
And instead of discussion, if we should avoid allocate/copy/deallocate and if yes, how to do it, I read comment about C style vs C++ style.
I don't understand the purpose of this comment.